### PR TITLE
Monitoring install scripts

### PR DESCRIPTION
**Pull request by freimair**
Completes first part of #3916.

- cleans up the seednode install script
- install script for monitoring servers with collectd
- install script for the networksize metric

I tested the script for general monitoring using a fresh ubuntu 18.04 LTS installation. I could not test the script for the networksize metric as thoroughly, as there is no install script for a price node available yet.

### CODE REVIEW

**Comment by ripcurlx** [comment]
@freimair Could you have a look at https://app.codacy.com/gh/bisq-network/bisq/pullRequest?prid=5023490. If there are false positives please let me know and I'll adapt the checks. Thanks!

**Comment by freimair** [comment]
> @freimair Could you have a look at https://app.codacy.com/gh/bisq-network/bisq/pullRequest?prid=5023490. If there are false positives please let me know and I'll adapt the checks. Thanks!

you beat me to that. I already fixed the issues. And nope, the checks seem fine. You get these issues if you copy and paste your way through life.

Is there a way to use codacy before creating the PR? (because no force-pushes, no fix commits, yadda yadda...)

**Comment by ripcurlx** [comment]
> Is there a way to use codacy before creating the PR? (because no force-pushes, no fix commits, yadda yadda...)

The only way would be if you enable it on your fork yourself to run on every branch you push.

**Comment by wiz** [comment]
@freimair see this line for a workaround for the codacy issue https://github.com/bisq-network/bisq-explorer/blob/master/install_bsq_explorer_debian.sh#L56

**Comment by ripcurlx** [comment]
> @freimair see this line for a workaround for the codacy issue https://github.com/bisq-network/bisq-explorer/blob/master/install_bsq_explorer_debian.sh#L56

If you guys think this Codacy remark is not correct, then I can just deactivate this single check.

**Comment by freimair** [comment]
should all be good now. My bad.

**Review by cbeams — COMMENTED** [review]
I was pinged here for a code review because I'm the `pricenode` code owner, but the changes there are minimal and do not affect production code, so I'm giving a my OK from the pricenode side, but am abstaining from a larger review.

File: monitor/install_collectd_debian.sh [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+read -p "Please provide the onion address of your service (eg. 3f3cu2yw7u457ztq): " onionaddress
+
+echo "[*] Updating apt repo sources"
+sudo -H -i -u "${ROOT_USER}" DEBIAN_FRONTEND=noninteractive apt-get update -q
+
+echo "[*] Upgrading OS packages"
+sudo -H -i -u "${ROOT_USER}" DEBIAN_FRONTEND=noninteractive apt-get upgrade -qq -y
+
+echo "[*] Installing base packages"
+sudo -H -i -u "${ROOT_USER}" DEBIAN_FRONTEND=noninteractive apt-get install -qq -y "${ROOT_PKG}"
```
Comment (wiz) [inline monitor/install_collectd_debian.sh]:
```suggestion
sudo -H -i -u "${ROOT_USER}" DEBIAN_FRONTEND=noninteractive apt-get install -qq -y ${ROOT_PKG}
```

File: monitor/install_collectd_debian.sh [hunk 2]
```diff
@@ -0,0 +1,62 @@
+#!/bin/sh
+set -e
+
+echo "[*] Bisq Server Monitoring installation script"
+
+##### change paths if necessary for your system
+
```
Comment (wiz) [inline monitor/install_collectd_debian.sh]:
```suggestion
BISQ_REPO_URL=https://raw.githubusercontent.com/bisq-network/bisq
BISQ_REPO_TAG=master
```

File: monitor/install_collectd_debian.sh (hunk 1 above)
Comment (wiz) [inline monitor/install_collectd_debian.sh]:
```suggestion
sudo -H -i -u "${ROOT_USER}" DEBIAN_FRONTEND=noninteractive apt-get install -qq -y ${ROOT_PKG}
```

File: monitor/install_collectd_debian.sh [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+sudo -H -i -u "${ROOT_USER}" DEBIAN_FRONTEND=noninteractive apt-get install -qq -y "${ROOT_PKG}"
+
+echo "[*] Preparing Bisq init script for monitoring"
+# remove stuff it it is there already
+sudo -H -i -u "${ROOT_USER}" sed -i 's/ -Dcom.sun.management.jmxremote -Dcom.sun.management.jmxremote.host=127.0.0.1 -Dcom.sun.management.jmxremote.port=6969 -Dcom.sun.management.jmxremote.rmi.port=6969 -Dcom.sun.management.jmxremote.ssl=false -Dcom.sun.management.jmxremote.authenticate=false//' "${SYSTEMD_ENV_HOME}/bisq.env"
+sudo -H -i -u "${ROOT_USER}" sed -i -e '/JAVA_OPTS/ s/"$/ -Dcom.sun.management.jmxremote -Dcom.sun.management.jmxremote.local.only=true -Dcom.sun.management.jmxremote.port=6969 -Dcom.sun.management.jmxremote.rmi.port=6969 -Dcom.sun.management.jmxremote.ssl=false -Dcom.sun.management.jmxremote.authenticate=false"/' "${SYSTEMD_ENV_HOME}/bisq.env"
+
+echo "[*] Installing Nginx config"
+sudo -H -i -u "${ROOT_USER}" openssl req -x509 -nodes -newkey rsa:2048 -keyout /etc/nginx/cert.key -out /etc/nginx/cert.crt -subj="/O=Bisq/OU=Bisq Infrastructure/CN=$onionaddress"
+curl -s https://raw.githubusercontent.com/bisq-network/bisq/master/monitor/nginx.conf > /tmp/nginx.conf
```
Comment (wiz) [inline monitor/install_collectd_debian.sh]:
```suggestion
curl -s "${BISQ_REPO_URL}/${BISQ_REPO_TAG}/monitor/nginx.conf" > /tmp/nginx.conf
```

File: monitor/install_collectd_debian.sh [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+sudo -H -i -u "${ROOT_USER}" install -c -o "${ROOT_USER}" -g "${ROOT_GROUP}" -m 644 /tmp/nginx.conf /etc/nginx/nginx.conf
+
+echo "[*] Installing collectd config"
+curl -s https://raw.githubusercontent.com/bisq-network/bisq/master/monitor/collectd.conf > /tmp/collectd.conf
```
Comment (wiz) [inline monitor/install_collectd_debian.sh]:
```suggestion
curl -s "${BISQ_REPO_URL}/${BISQ_REPO_TAG}/monitor/collectd.conf" > /tmp/collectd.conf
```

File: monitor/install_collectd_debian.sh [hunk 5]
```diff
@@ -0,0 +1,62 @@
+#!/bin/sh
+set -e
+
+echo "[*] Bisq Server Monitoring installation script"
+
+##### change paths if necessary for your system
+
+ROOT_USER=root
+ROOT_GROUP=root
```
Comment (wiz) [inline monitor/install_collectd_debian.sh:10]:
```suggestion
ROOT_GROUP=root
ROOT_HOME=~root
```

File: monitor/install_collectd_debian.sh [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+sudo -H -i -u "${ROOT_USER}" DEBIAN_FRONTEND=noninteractive apt-get upgrade -qq -y
+
+echo "[*] Installing base packages"
+sudo -H -i -u "${ROOT_USER}" DEBIAN_FRONTEND=noninteractive apt-get install -qq -y "${ROOT_PKG}"
+
+echo "[*] Preparing Bisq init script for monitoring"
+# remove stuff it it is there already
+sudo -H -i -u "${ROOT_USER}" sed -i 's/ -Dcom.sun.management.jmxremote -Dcom.sun.management.jmxremote.host=127.0.0.1 -Dcom.sun.management.jmxremote.port=6969 -Dcom.sun.management.jmxremote.rmi.port=6969 -Dcom.sun.management.jmxremote.ssl=false -Dcom.sun.management.jmxremote.authenticate=false//' "${SYSTEMD_ENV_HOME}/bisq.env"
+sudo -H -i -u "${ROOT_USER}" sed -i -e '/JAVA_OPTS/ s/"$/ -Dcom.sun.management.jmxremote -Dcom.sun.management.jmxremote.local.only=true -Dcom.sun.management.jmxremote.port=6969 -Dcom.sun.management.jmxremote.rmi.port=6969 -Dcom.sun.management.jmxremote.ssl=false -Dcom.sun.management.jmxremote.authenticate=false"/' "${SYSTEMD_ENV_HOME}/bisq.env"
+
```
Comment (wiz) [inline monitor/install_collectd_debian.sh:34]:
```suggestion

echo "[*] Seeding entropy from /dev/urandom"
sudo -H -i -u "${ROOT_USER}" /bin/sh -c "head -1500 /dev/urandom > ${ROOT_HOME}/.rnd"
```

**Review by wiz — CHANGES_REQUESTED** [review]

File: monitor/install_collectd_debian.sh [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+echo "[*] Upgrading OS packages"
+sudo -H -i -u "${ROOT_USER}" DEBIAN_FRONTEND=noninteractive apt-get upgrade -qq -y
+
+echo "[*] Installing base packages"
+sudo -H -i -u "${ROOT_USER}" DEBIAN_FRONTEND=noninteractive apt-get install -qq -y "${ROOT_PKG}"
+
+echo "[*] Preparing Bisq init script for monitoring"
+# remove stuff it it is there already
+sudo -H -i -u "${ROOT_USER}" sed -i 's/ -Dcom.sun.management.jmxremote -Dcom.sun.management.jmxremote.host=127.0.0.1 -Dcom.sun.management.jmxremote.port=6969 -Dcom.sun.management.jmxremote.rmi.port=6969 -Dcom.sun.management.jmxremote.ssl=false -Dcom.sun.management.jmxremote.authenticate=false//' "${SYSTEMD_ENV_HOME}/bisq.env"
```
Comment (wiz) [inline monitor/install_collectd_debian.sh]:
```suggestion
sudo -H -i -u "${ROOT_USER}" sed -i -e 's/ -Dcom.sun.management.jmxremote //g' -e 's/-Dcom.sun.management.jmxremote.local.only=true//g' -e 's/ -Dcom.sun.management.jmxremote.host=127.0.0.1//g' -e 's/ -Dcom.sun.management.jmxremote.port=6969//g' -e 's/ -Dcom.sun.management.jmxremote.rmi.port=6969//g' -e 's/ -Dcom.sun.management.jmxremote.ssl=false//g' -e 's/ -Dcom.sun.management.jmxremote.authenticate=false//g' "${SYSTEMD_ENV_HOME}/bisq.env"
```

File: monitor/install_collectd_debian.sh (hunk 5 above)
Comment (freimair) [reply monitor/install_collectd_debian.sh:10]: ROOT_HOME is not used in the script

Comment (wiz) [reply monitor/install_collectd_debian.sh:10]: It's used in the entropy seeding I added

File: monitor/install_collectd_debian.sh [hunk 8]
```diff
@@ -0,0 +1,66 @@
+#!/bin/sh
```
Comment (wiz) [inline monitor/install_collectd_debian.sh]:
```suggestion
#!/bin/bash
```

File: monitor/install_collectd_debian.sh [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+
+echo "[*] Bisq Server Monitoring installation script"
+
+##### change paths if necessary for your system
+BISQ_REPO_URL=https://raw.githubusercontent.com/bisq-network/bisq
+BISQ_REPO_TAG=master
+ROOT_USER=root
+ROOT_GROUP=root
+ROOT_HOME=~root
+ROOT_PKG="nginx collectd openssl"
```
Comment (wiz) [inline monitor/install_collectd_debian.sh]:
```suggestion
ROOT_PKG=(nginx collectd openssl)
```

File: monitor/install_collectd_debian.sh [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+read -p "Please provide the onion address of your service (eg. 3f3cu2yw7u457ztq): " onionaddress
+
+echo "[*] Updating apt repo sources"
+sudo -H -i -u "${ROOT_USER}" DEBIAN_FRONTEND=noninteractive apt-get update -q
+
+echo "[*] Upgrading OS packages"
+sudo -H -i -u "${ROOT_USER}" DEBIAN_FRONTEND=noninteractive apt-get upgrade -qq -y
+
+echo "[*] Installing base packages"
+sudo -H -i -u "${ROOT_USER}" DEBIAN_FRONTEND=noninteractive apt-get install -qq -y ${ROOT_PKG}
```
Comment (wiz) [inline monitor/install_collectd_debian.sh]:
```suggestion
sudo -H -i -u "${ROOT_USER}" DEBIAN_FRONTEND=noninteractive apt-get install -qq -y ${ROOT_PKG[@]}
```

File: monitor/install_collectd_debian.sh [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+
+echo "[*] Preparing Bisq init script for monitoring"
+# remove stuff it it is there already
+sudo -H -i -u "${ROOT_USER}" sed -i -e 's/ -Dcom.sun.management.jmxremote //g' -e 's/-Dcom.sun.management.jmxremote.local.only=true//g' -e 's/ -Dcom.sun.management.jmxremote.host=127.0.0.1//g' -e 's/ -Dcom.sun.management.jmxremote.port=6969//g' -e 's/ -Dcom.sun.management.jmxremote.rmi.port=6969//g' -e 's/ -Dcom.sun.management.jmxremote.ssl=false//g' -e 's/ -Dcom.sun.management.jmxremote.authenticate=false//g' "${SYSTEMD_ENV_HOME}/bisq.env"
```
Comment (wiz) [inline monitor/install_collectd_debian.sh]:
```suggestion
sudo -H -i -u "${ROOT_USER}" sed -i -e 's/-Dcom.sun.management.jmxremote //g' -e 's/-Dcom.sun.management.jmxremote.local.only=true//g' -e 's/ -Dcom.sun.management.jmxremote.host=127.0.0.1//g' -e 's/ -Dcom.sun.management.jmxremote.port=6969//g' -e 's/ -Dcom.sun.management.jmxremote.rmi.port=6969//g' -e 's/ -Dcom.sun.management.jmxremote.ssl=false//g' -e 's/ -Dcom.sun.management.jmxremote.authenticate=false//g' "${SYSTEMD_ENV_HOME}/bisq.env"
```

**Review by wiz — CHANGES_REQUESTED** [review]

**Review by wiz — APPROVED** [review]
Tested :+1: on fresh Ubuntu 18 install using seednode installer + collectd installer

**Review by ripcurlx — APPROVED** [review]
utACK

Based on https://github.com/bisq-network/bisq/pull/3985#pullrequestreview-361921522

**Pull request merged**